src/main/kotlin/no/nav/familie/ef/sak/vedtak/AndelHistorikkBeregner.kt

```diff
@@ -49,7 +50,36 @@ object AndelHistorikkBeregner {
 
     fun lagHistorikk(tilkjentYtelser: List<TilkjentYtelse>,
                      vedtaksliste: List<Vedtak>,
-                     behandlinger: List<Behandling>): List<AndelHistorikkDto> {
+                     behandlinger: List<Behandling>,
+                     tilOgMedBehandlingId: UUID?): List<AndelHistorikkDto> {
+        return if (tilOgMedBehandlingId == null) {
+            lagHistorikk(tilkjentYtelser, vedtaksliste, behandlinger)
+        } else {
+            lagHistorikkTilBehandlingId(tilkjentYtelser, vedtaksliste, behandlinger, tilOgMedBehandlingId)
+        }
+    }
+
+    /**
+     * Filtrerer vekk data som kommer etter behandlingen som man sender inn
+     */
+    private fun lagHistorikkTilBehandlingId(tilkjentYtelser: List<TilkjentYtelse>,
+                                            vedtaksliste: List<Vedtak>,
+                                            behandlinger: List<Behandling>,
+                                            tilOgMedBehandlingId: UUID?): List<AndelHistorikkDto> {
```

Comment:
Denne kan vel egentlig ikke være null?

Reply:
Jeg tenkte det var greit hvis den var null, og att den då ikke filtrerer bort tidligere behandlinger. Sånn att når man ikke valgt noe i frontend (eller ja, valgt siste) så sender man null. Også litt for å håndtere att backend merges før frontend :)